ext/drbbdup/drbbdup.c

```diff
@@ -1743,10 +1744,12 @@ drbbdup_prepare_redirect(dr_mcontext_t *mcontext, drbbdup_manager_t *manager,
                          app_pc bb_pc)
 {
     /* Restore flags and scratch reg to their original app values. */
+#if !defined(RISCV64)
```

Comment:
What about the declaration of the `are_flags_dead` field and the handful of other references: should that all be ifdef-ed out as well?  Just this one and not others seems unusual?

Reply:
I wanted to err on the side of caution and leave the scratch register here. I'm sure I won't use the flags register, though I'm not so sure about the scratch register.

Reply:
> I wanted to err on the side of caution and leave the scratch register here.

Are you referring to `is_scratch_reg_dead`?  That's a separate field: I'm only talking about the other references to `are_aflags_dead` in this and the .h file.

Reply:
Correct me if I'm wrong but that would mean commenting out `drreg_reserve_aflags()`, `drreg_unreserve_aflags()`, `drreg_aflags_liveness()` and `drreg_are_aflags_dead()`? Then any related functionality such as `drx_aflags_are_dead()` or logic in `tracer.c`?

Reply:
OK I have disabled this code though I'm not sure if I'm not missing some other APIs.

Reply:
No I mean only the `are_aflags_dead` field in the `drbbdup_per_thread` structure in drbbup.c (I thought it was in drbbdup.h too but maybe not).  The PR had an ifdef around one reference to that field, but there are a couple more, so it makes sense to disable all of those.  I didn't mean all the aflags interfaces in other files.

Reply:
OK, I got confused when you've mentioned `are_aflags_dead` and a header file, whereas the field in `drbbdup_manager_t` is called `are_flags_dead` and it's in the `drbbdup.c` file.
I've reverted interface changes and left out ifdefs around the usage of `drbbdup_manager_t::are_flags_dead`.

Reply:
Sorry, I thought I was being precise with the backtics but I typed it wrong and didn't look closely at the `git grep` to see it was all in the .c file.